modules/migrating-between-capi-mapi.adoc

```diff
@@ -0,0 +1,153 @@
+// Module included in the following assemblies:
+//
+// * machine_management/cluster_api_machine_management/cluster-api-disabling.adoc
+// * machine_management/cluster_api_machine_management/cluster-api-getting-started.adoc
+
+ifeval::["{context}" == "cluster-api-getting-started"]
+:machine-to-cluster:
+endif::[]
+ifeval::["{context}" == "cluster-api-disabling"]
+:cluster-to-machine:
+endif::[]
+
+ifdef::machine-to-cluster[]
+:from-api-name: Machine API
+:to-api-name: Cluster API
+:from-api-value: MachineAPI
+:to-api-value: ClusterAPI
+:from-api-group: machine.openshift.io
+:to-api-group: cluster.x-k8s.io
+:origin-namespace: openshift-machine-api
+endif::[]
+
+ifdef::cluster-to-machine[]
+:from-api-name: Cluster API
+:to-api-name: Machine API
+:from-api-value: ClusterAPI
+:to-api-value: MachineAPI
+:from-api-group: cluster.x-k8s.io
+:to-api-group: machine.openshift.io
+:origin-namespace: openshift-cluster-api
+endif::[]
+
+:_mod-docs-content-type: PROCEDURE
+[id="migrating-between-capi-mapi_{context}"]
+= Migrating a {from-api-name} resource to use the {to-api-name}
+
+You can migrate individual {from-api-name} objects to equivalent {to-api-name} objects.
+
+:FeatureName: Migrating a {from-api-name} resource to use the {to-api-name}
+include::snippets/technology-preview.adoc[]
+
+.Prerequisites
+
+* You have deployed an {product-title} cluster on a supported infrastructure type.
+
+ifdef::machine-to-cluster[]
+* You have enabled the use of the Cluster API.
+endif::[]
+
+* You have enabled the `MachineAPIMigration` feature gate in the `TechPreviewNoUpgrade` feature set.
+
+* You have access to the cluster using an account with `cluster-admin` permissions.
+
+* You have installed the {oc-first}.
+
+.Procedure
+
+. Identify the {from-api-name} resource that you want to migrate to a {to-api-name} resource by running the following command:
++
+[source,terminal,subs="attributes+"]
+----
+$ oc get <resource_kind> -n {origin-namespace}
+----
++
+--
+where `<resource_kind>` is one of the following values:
+
+`machine.{from-api-group}`:: The fully qualified name of the resource kind for a compute or control plane machine.
++
+[IMPORTANT]
+====
+The output of this command includes the control plane machines.
+Migrating a control plane machine in not supported.
+====
+
+`machineset.{from-api-group}`:: The fully qualified name of the resource kind for a compute machine set.
+--
+
+. Edit the resource specification by running the following command:
++
+[source,terminal]
+----
+$ oc edit <resource_name>
+----
++
+where `<resource_name>` is the fully qualified name of the {from-api-name} resource that you want to migrate to a {to-api-name} resource.
```

Comment:
This part will cause some wrong if the `{from-api-name}` is `Cluster API`, because we can only update the value of the `spec.authoritativeAPI` field for the `Machine API` resource, there is no `spec.authoritativeAPI` field for the `Cluster API` resource.

Reply:
Each authoritative MAPI resource have a non-authoritative mirror CAPI resource  and each authoritative CAPI resource have a non-authoritative mirror MAPI resource. 
If they want to migrate a MAPI resource to CAPI, they need to change the MAPI resource `spec.authoritativeAPI` to ClusterAPI. If they want to migrate a CAPI resource to MAPI, they need to change the equivalent MAPI resource `spec.authoritativeAPI` to MachineAPI.
So no matter they want to migrate from MAPI to CAPI, or form CAPI to MAPI, they need update the same place (the MAPI resource `spec.authoritativeAPI`), but to different values (ClusterAPI or MachineAPI).